localstack-core/localstack/services/apigateway/provider.py

```diff
@@ -428,7 +428,10 @@ def create_domain_name(
 
         # find matching hosted zone
         zone_id = None
-        route53 = connect_to().route53
+        # TODO check if this call is IAM enforced
+        route53 = connect_to(
+            region_name=context.region, aws_access_key_id=context.account_id
+        ).route53
```

Comment:
Yeah, this is pretty interesting. I'd say it might not get enforced, as I'm not sure we'd have any way to pass credentials here. 

I'm not too aware of the details of the domain name creation and logic, but from the documentation, it seems we're retrieving the `regionalHostedZoneId` but this is the doc:
> [distributionHostedZoneId](https://docs.aws.amazon.com/apigateway/latest/api/API_CreateDomainName.html#API_CreateDomainName_ResponseSyntax)
The region-agnostic Amazon Route 53 Hosted Zone ID of the edge-optimized endpoint. The valid value is Z2FDTNDATAQYW2 for all the regions. For more information, see Set up a Regional Custom Domain Name and AWS Regions and Endpoints for API Gateway.

See [the doc page](https://docs.aws.amazon.com/apigateway/latest/api/API_CreateDomainName.html).
So it is a bit weird 😄

Reply:
It might be global (not sure if that lookup works though), but it is definitely only partition-global, so hitting the right partition should definitely be our goal here - might as well use the right region then :)

Reply:
Oh sorry, I meant this comment about the `# TODO check if this call is IAM enforced`, the change is definitely good! 😅